process.c

```diff
@@ -8779,16 +8779,58 @@ proc_warmup(VALUE _)
  *
  * == \Process Creation
  *
- * Each of these methods creates a process:
- *
- * - Process.exec: Replaces the current process by running a given external command.
- * - Process.spawn, Kernel#spawn: Executes the given command and returns its pid without waiting for completion.
- * - Kernel#system: Executes the given command in a subshell.
- *
- * Each of these methods accepts:
- *
- * - An optional hash of environment variable names and values.
- * - An optional hash of execution options.
+ * Each of the following methods executes a given command in a new process or subshell,
+ * or multiple commands in new processes and/or subshells.
+ * The choice of process or subshell depends on the form of the command;
+ * see {Argument command_line or exe_path}[rdoc-ref:Process@Argument+command_line+or+exe_path].
+ *
+ * - Each of these methods executes a single command in a process or subshell,
+ *   accepts a string for input to $stdin,
+ *   and returns string output from $stdout, $stderr, or both:
+ *
+ *   - Open3.capture2: Executes the command;
+ *     returns the string from $stdout.
+ *   - Open3.capture2e: Executes the command;
+ *     returns the string from merged $stdout and $stderr.
+ *   - Open3.capture3: Executes the command;
+ *     returns strings from $stdout and $stderr.
+ *
+ * - Each of these methods executes a single command in a process or subshell,
+ *   and returns pipes for $stdin, $stdout, and/or $stderr:
+ *
+ *   - Open3.popen2: Executes the command;
+ *     returns pipes for $stdin and $stdout.
+ *   - Open3.popen2e: Executes the command;
+ *     returns pipes for $stdin and merged $stdout and $stderr.
+ *   - Open3.popen3: Executes the command;
+ *     returns pipes for $stdin, $stdout, and $stderr.
+ *
+ * - Each of these methods executes one or more commands in processes and/or subshells,
+ *   returns pipes for the first $stdin, the last $stdout, or both:
+ *
+ *   - Open3.pipeline_r: Returns a pipe for the last $stdout.
+ *   - Open3.pipeline_rw: Returns pipes for the first $stdin and the last $stdout.
+ *   - Open3.pipeline_w: Returns a pipe for the first $stdin.
+ *   - Open3.pipeline_start: Does not wait for processes to complete.
+ *   - Open3.pipeline: Waits for processes to complete.
```

Comment:
`Open3` is not part of `Process`, so I don't think this should belong in the documentation for `Process`.

Reply:
I tend to agree, but I think somewhere we should be giving this overview of process creation, to help the users understand and choose.  I'd thought about doc/process.rdoc as a home for this, but the search engines love our classes and hate our doc pages.  Should this material be in doc/ anyway?

Reply:
Rather than going into details about `Open3`, I think this level of detail should belong in `Open3` and then in `Process` just briefly state that `Open3` has convenient methods for shelling out.

Reply:
Done;  Kernel#` mention also added.